### PR TITLE
Fixes issue of odo login not working correctly

### PR DESCRIPTION
**Pull request by cdrage**
This fixes the issue of `odo login` not working correctly.

In this PR:
  - Added logging for before AND after
  - Changed stdout / stderror to the correct ones from odolog package
  - Fix up success output
  - Added Success function to odolog

Closes issue https://github.com/openshift/odo/issues/1643
Closes issue https://github.com/openshift/odo/issues/1664
Closes issue https://github.com/openshift/odo/issues/1662

### CODE REVIEW

**Comment by cdrage** [comment]
What the changes look like:

[asciicast](https://asciinema.org/a/EFv0YcWGO74Z0XyIDSRcNOebG)

**Comment by kadel** [comment]
not buildable 
```
pkg/auth/login.go:5:2: imported and not used: "fmt"
```

**Comment by girishramnani** [comment]
after a failure the color still stay's yellow
Image Pasted at 2019-4-26 11-40

**Comment by amitkrout** [comment]
@cdrage 
```
CDKQE@DESKTOP-N3T8HBG MINGW64 ~/go/src/github.com/openshift/odo (pull_1652)
$ ./odo.exe login -u kubeadmin -p password https://console-openshift-console.apps.amit33.devcluster.openshift.com
Connecting to the OpenShift cluster

The server uses a certificate signed by an unknown authority.
You can bypass the certificate check, but any data you send to the server could be intercepted by others.
Use insecure connections? (y/n): y


 X  invalid character '<' looking for beginning of value
```

**Comment by cdrage** [comment]
@amitkrout I get the same error with `oc`, looks like an issue with your cluster?

```sh
github.com/openshift/odo  add-show-log-to-watch ✔                                                                                                                                                                                                                                                                                                                     10m  ⍉
▶ oc login -u kubeadmin -p password https://console-openshift-console.apps.amit33.devcluster.openshift.com      
The server uses a certificate signed by an unknown authority.
You can bypass the certificate check, but any data you send to the server could be intercepted by others.
Use insecure connections? (y/n): y

error: invalid character '<' looking for beginning of value
```

Can you please do another review / LGTM? :)

**Comment by amitkrout** [comment]
> @amitkrout I get the same error with `oc`, looks like an issue with your cluster?
> 
> ```shell
> github.com/openshift/odo  add-show-log-to-watch ✔                                                                                                                                                                                                                                                                                                                     10m  ⍉
> ▶ oc login -u kubeadmin -p password https://console-openshift-console.apps.amit33.devcluster.openshift.com      
> The server uses a certificate signed by an unknown authority.
> You can bypass the certificate check, but any data you send to the server could be intercepted by others.
> Use insecure connections? (y/n): y
> 
> error: invalid character '<' looking for beginning of value
> ```
> 
> Can you please do another review / LGTM? :)

I have seen such error message ```X  invalid character '<' looking for beginning of value```. In my case what I observed that it happens only if the client version is not compatible with the running cluster

For example :
My cluster version is 4.1 running on aws
And I am trying to login the cluster with 3.11 client version

@kadel can you please check it against your cluster.

**Comment by amitkrout** [comment]
@cdrage Even i am seeing the same error while using latest 4.1 oc client for windows [1]. Seems its an issue on 4.1. 

[1] https://mirror.openshift.com/pub/openshift-v3/clients/4.1.0-201904090034.git.0.1016bda.el7/windows/

**Comment by cdrage** [comment]
@amitkrout @girishramnani Yup, looks like an error outside-of-Odo.

Mind doing a quick review of this PR?

**Comment by amitkrout** [comment]
> @amitkrout @girishramnani Yup, looks like an error outside-of-Odo.
> 
> Mind doing a quick review of this PR?

@cdrage When i check odo login without this fix it works fine on minishift 1.33.0 (oc 3.11.0). The only issue i found is odo login is broken if trying to login 4.1 cluster. 

This pr fix the login issue on 4.1, however odo logout is broken even on master too.

```
C:\Users\user\minishift-1.33.0-windows-amd64>odo.exe login
Connecting to the OpenShift cluster

Authentication required for https://192.168.99.100:8443 (openshift)
Username: developer
Password:
Login successful.

You have one project on this server: "myproject"

Using project "myproject".

 V  Login succeeded
```
```
C:\Users\user\Downloads\minishift-1.33.0-windows-amd64>odo.exe logout
 X  the current directory does not represent an odo component.
Maybe use 'odo create' to create component here or switch to directory with a component

```

**Comment by cdrage** [comment]
@amitkrout Okay, I'll fix this in a separate PR. This PR will only fix `odo login`. I'll add the fix for `odo logout` in a separate PR.

**Comment by kadel** [comment]
/retest

```
error: You must be logged in to the server (the server has asked for the client to provide credentials)
```

**Comment by cdrage** [comment]
Fixed another error pertaining to this issue here: https://github.com/openshift/odo/issues/1664

Ready for another review @surajnarwade @kadel @girishramnani @amitkrout

**Comment by kadel** [comment]
It looks like something is broken in `odo login`

```
  /home/travis/gopath/src/github.com/openshift/odo/tests/e2e/login_test.go:37
Running command: odo login -t 
Error: flag needs an argument: 't' in -t
```

**Comment by kadel** [comment]
This fixes both login issues, which is great.

Problem is that it reverts back to the original messages from the `oc` :-(

```
▶ odo login
Connecting to the OpenShift cluster

Server [https://localhost:8443]: https://api.tkral.devcluster.openshift.com:6443
The server uses a certificate signed by an unknown authority.
You can bypass the certificate check, but any data you send to the server could be intercepted by others.
Use insecure connections? (y/n): y

Authentication required for https://api.tkral.devcluster.openshift.com:6443 (openshift)
Username: developer
Password:
Login successful.

You don't have any projects. You can try to create a new project, by running

    odo new-project <projectname>


 ✓  Login succeeded
````

````
▶ odo login https://mycluster:6443
Connecting to the OpenShift cluster

The server uses a certificate signed by an unknown authority.
You can bypass the certificate check, but any data you send to the server could be intercepted by others.
Use insecure connections? (y/n): y

Authentication required for https://mycluster:6443 (openshift)
Username: kubeadmin
Password:
Login successful.

You have access to the following projects and can switch between them with 'odo project <projectname>':

  * default
    kube-public
    kube-system
    myproject

Using project "default".

 ✓  Login succeeded
````


`odo project <projectname>` and `odo new-project <projectname>` are not valid odo commands. It sahould say `odo project set <projectname>` and `odo project create <projectname>`

**Comment by cdrage** [comment]
> This fixes both login issues, which is great.
> 
> Problem is that it reverts back to the original messages from the `oc` :-(
> 
> ```
> ▶ odo login
> Connecting to the OpenShift cluster
> 
> Server [https://localhost:8443]: https://api.tkral.devcluster.openshift.com:6443
> The server uses a certificate signed by an unknown authority.
> You can bypass the certificate check, but any data you send to the server could be intercepted by others.
> Use insecure connections? (y/n): y
> 
> Authentication required for https://api.tkral.devcluster.openshift.com:6443 (openshift)
> Username: developer
> Password:
> Login successful.
> 
> You don't have any projects. You can try to create a new project, by running
> 
>     odo new-project <projectname>
> 
> 
>  ✓  Login succeeded
> ```
> 
> ```
> ▶ odo login https://mycluster:6443
> Connecting to the OpenShift cluster
> 
> The server uses a certificate signed by an unknown authority.
> You can bypass the certificate check, but any data you send to the server could be intercepted by others.
> Use insecure connections? (y/n): y
> 
> Authentication required for https://mycluster:6443 (openshift)
> Username: kubeadmin
> Password:
> Login successful.
> 
> You have access to the following projects and can switch between them with 'odo project <projectname>':
> 
>   * default
>     kube-public
>     kube-system
>     myproject
> 
> Using project "default".
> 
>  ✓  Login succeeded
> ```
> 
> `odo project <projectname>` and `odo new-project <projectname>` are not valid odo commands. It sahould say `odo project set <projectname>` and `odo project create <projectname>`

I'm going to manipulate stdout / stderr in order to achieve a better form of filtering for this (and the future). Thanks for pointing this out!

**Comment by cdrage** [comment]
Up for another review @girishramnani @kadel @surajnarwade @mik-dass @amitkrout

**Comment by mik-dass** [comment]
If I press `crtl+c` in between the inputs, the colour of the terminal stays yellow

**Comment by cdrage** [comment]
@mik-dass Thanks!

I've gone ahead and removed the coloured output. It was causing too many problems.

**Comment by girishramnani** [comment]
\lgtm

File: pkg/auth/login.go [hunk 1]
```diff
@@ -47,16 +58,76 @@ func Login(server, username, password, token, caAuth string, skipTLS bool) error
...
+}
+
+// filteredInformation takes a list of strings ([]byte), replaces them and spits it back out
+// This is used since we utilize `oc login` withi odo and require certain strings to be filtered / changed
+// to their odo equivilant
+func filteredInformation(s []byte) []byte {
+
+	// List of strings to correctly filter
+	s = bytes.Replace(s, []byte("new-project"), []byte("project create"), -1)
+	s = bytes.Replace(s, []byte("<projectname>"), []byte("<project-name>"), -1)
```
Comment (kadel) [inline pkg/auth/login.go]: Why we are replacing `projectname` with `project-name`?

**Comment by kadel** [comment]
There is still one more thing to modify in the output.

`You have access to the following projects and can switch between them with 'odo project <project-name>'`

should be `odo project set <project-name>`

File: pkg/auth/login.go (hunk 1 above)
Comment (cdrage) [reply pkg/auth/login.go]: That's how it was before  in the previous code, so no idea to be honest as to why people chose that way, preference?

**Comment by kadel** [comment]
> There is still one more thing to modify in the output.
> 
> `You have access to the following projects and can switch between them with 'odo project <project-name>'`
> 
> should be `odo project set <project-name>`

ping @cdrage

**Comment by cdrage** [comment]
@kadel Done!

**Comment by cdrage** [comment]
Going to re-base this against master as updated tests have been merged :+1:

**Comment by cdrage** [comment]
Rebased ^^. Time for another review :+1:

**Pull request merged**